core/src/main/scala/tamer/Tamer.scala

```diff
@@ -43,24 +43,24 @@ object Tamer {
   }
 
   private[tamer] final def sink[K, V](
-      stream: UStream[(K, V)],
-      producer: Producer,
+      stream: UStream[(Transaction, K, V)],
       topic: String,
       keySerializer: Serializer[Has[Registry], K],
       valueSerializer: Serializer[Has[Registry], V],
       log: LogWriter[Task]
   ) =
     stream
-      .map { case (k, v) => new ProducerRecord(topic, k, v) }
+//      .map { case (k, v) => new ProducerRecord(topic, k, v) }
       .mapChunksM {
-        case chunk if chunk.nonEmpty =>
-          producer
-            .produceChunk(chunk, keySerializer, valueSerializer)
+        case chunk if chunk.nonEmpty && chunk.forall { case (transaction, _, _) => transaction == chunk.head._1 } =>
+          val transaction = chunk.head._1
+          transaction
+            .produceChunk(chunk.map { case (_, k, v) => new ProducerRecord(topic, k, v) }, keySerializer, valueSerializer, None)
```

Comment:
This is part of the transaction, *BUT* if a consumer is interested in having this as soon as possible and tolerates duplicates, it can just `read_uncommitted`. Does this make sense?